docs/ref/templates/language.txt

```diff
@@ -79,9 +79,9 @@ Variables
 Variables look like this: ``{{ variable }}``. When the template engine
 encounters a variable, it evaluates that variable and replaces it with the
 result. Variable names consist of any combination of alphanumeric characters
-and the underscore (``"_"``) but may not start with an underscore. The dot
-(``"."``) also appears in variable sections, although that has a special
-meaning, as indicated below. Importantly, *you cannot have spaces or
+and the underscore (``"_"``) but may not start with an underscore or a number.
```

Comment:
It can start with a number but cannot be a number, so maybe:
```suggestion
and the underscore (``"_"``) but may not start with an underscore and cannot be a number.
```

Reply:
Thanks :) I've made that change.

Yes - I wonder if it's worth not allowing variable names that start with a number anyway. (Not actually blocking them, just defining here that we don't accept them). Whilst they wouldn't break anything, it would mean that DTL variable names would just be python variable names that can't start with an underscore. 🤔

I guess that would require a discussion and a decision though, so for another day/ticket.

Reply:
I think it's enough to document the current state.